src/Dapr.Actors.AspNetCore/ActorsServiceCollectionExtensions.cs

```diff
@@ -72,5 +93,88 @@ public static void AddActors(this IServiceCollection services, Action<ActorRunti
+        /// <returns>The built gRPC endpoint.</returns>
+        private static string GetHttpEndpoint(IConfiguration? configuration)
+        {
+            //Prioritize pulling from IConfiguration with a fallback from pulling from the environment variable directly
+            var httpEndpoint = GetResourceValue(configuration, DaprDefaults.DaprHttpEndpointName);
+            var httpPort = GetResourceValue(configuration, DaprDefaults.DaprHttpPortName);
+            int? parsedGrpcPort = string.IsNullOrWhiteSpace(httpPort) ? null : int.Parse(httpPort);
+
+            var endpoint = BuildEndpoint(httpEndpoint, parsedGrpcPort);
+            return string.IsNullOrWhiteSpace(endpoint) ? $"http://localhost:{DaprDefaults.DefaultHttpPort}/" : endpoint;
+        }
+
+        /// <summary>
+        /// Retrieves the specified value prioritizing pulling it from <see cref="IConfiguration"/>, falling back
+        /// to an environment variable, and using an empty string as a default.
+        /// </summary>
+        /// <param name="configuration">An instance of an <see cref="IConfiguration"/>.</param>
+        /// <param name="name">The name of the value to retrieve.</param>
+        /// <returns>The value of the resource.</returns>
+        private static string GetResourceValue(IConfiguration? configuration, string name)
+        {
+            //Attempt to retrieve first from the configuration
+            var configurationValue = configuration?.GetValue<string?>(name);
+            if (configurationValue is not null)
+                return configurationValue;
+
+            //Fall back to the environment variable with the same name or default to an empty string
+            var envVar = Environment.GetEnvironmentVariable(name);
+            return envVar ?? string.Empty;
+        }
+
+        /// <summary>
+        /// Builds the endpoint provided an optional endpoint and optional port.
+        /// </summary>
+        /// <remarks>
+        /// Marked as internal for testing purposes.
+        /// </remarks>
+        /// <param name="endpoint">The endpoint.</param>
+        /// <param name="endpointPort">The port</param>
+        /// <returns>A constructed endpoint value.</returns>
+        internal static string BuildEndpoint(string? endpoint, int? endpointPort)
+        {
+            if (string.IsNullOrWhiteSpace(endpoint) && endpointPort is null)
+                return string.Empty;
+
+            var endpointBuilder = new UriBuilder();
+            if (!string.IsNullOrWhiteSpace(endpoint))
+            {
+                //Extract the scheme, host and port from the endpoint
+                var uri = new Uri(endpoint);
+                endpointBuilder.Scheme = uri.Scheme;
+                endpointBuilder.Host = uri.Host;
+                endpointBuilder.Port = uri.Port;
+
+                //Update the port if provided separately
+                if (endpointPort is not null)
```

Comment:
If the endpoint has been set, the port should be ignored.  (Having a single implementation would help reduce the chance of variation.)

Reply:
I could have sworn I removed this in a commit last night.

Reply:
Removed - refactoring everything to the common DaprDefaults now

Reply:
Ah, I did have it in a commit in the AspNetCore project, not actors. Both just point to DaprDefaults now.